src/store/scan/scan.effects.ts

```diff
@@ -196,6 +208,138 @@ const goToPayPro =
     }
   };
 
+const handleUnlock =
+  (data: string): Effect =>
+  async dispatch => {
+    const invoiceId = data.split('i/')[1].split('?')[0];
+    const network = data.includes('test') ? Network.testnet : Network.mainnet;
+    const result = await dispatch(unlockInvoice(invoiceId, network));
+
+    if (result === 'unlockSuccess') {
+      dispatch(goToPayPro(data));
+      return;
+    }
+
+    const {host} = new URL(GetPayProUrl(data));
+    try {
+      const invoice = await axios.get(
+        `https://${host}/invoiceData/${invoiceId}`,
+      );
+      if (invoice) {
+        dispatch(goToPayPro(data));
+        return;
+      }
+    } catch {}
+    switch (result) {
+      case 'pairingRequired':
+        navigationRef.navigate('Auth', {
+          screen: 'Login',
+          params: {
+            onLoginSuccess: () => {
+              const parentNav = navigationRef.getParent();
+
+              if (parentNav?.canGoBack()) {
+                parentNav.goBack();
+              } else {
+                navigationRef.navigate('Tabs', {screen: 'Home'});
+              }
+
+              dispatch(incomingData(data));
+            },
+          },
+        });
+        break;
+
+      // needs verification - send to bitpay id verify
+      case 'userShopperNotFound':
+      case 'tierNotMet':
+        dispatch(
+          showBottomNotificationModal({
+            type: 'warning',
+            title: t('Connect Your BitPay ID'),
+            enableBackdropDismiss: false,
+            message: t(
+              'To complete this payment, please login with your BitPay ID.',
+            ),
+            actions: [
+              {
+                text: t('CONTINUE'),
+                action: () => {
+                  Linking.openURL(
+                    `https://${host}/id/verify?context=unlockAppV&id=${invoiceId}`,
+                  );
+                },
+              },
+              {
+                text: t('Cancel'),
+                action: () => {},
+              },
+            ],
+          }),
+        );
+        break;
+      default:
+        dispatch(showBottomNotificationModal(GeneralError()));
+        break;
+    }
+  };
+
+const unlockInvoice =
+  (invoiceId: string, network: Network): Effect<Promise<string>> =>
+  async (dispatch, getState) => {
+    const {BITPAY_ID} = getState();
+    const token = BITPAY_ID.apiToken[network];
```

Comment:
This may not exist, eg. app is set to mainnet but you scan a testnet invoice or vice versa

How do we want to handle this? Maybe check network !== APP.network and display an error message?

Reply:
Following up on this, it returns `pairingRequired` but since the invoice is unclaimed it sends me to the confirm screen anyway, which may be ok? But still seems like a hole that needs to be plugged

Reply:
When it returns `pairingRequired` we make another call to check if we can fetch invoiceData 
- on success redirects to confirm screen
- on failure goes to the login/verification flow (401 if amount > 3000).

and yes we should add that check with error msg, will address that